backend/uasmartsignage/src/main/java/deti/uas/uasmartsignage/Services/FileService.java

```diff
@@ -186,71 +181,196 @@ public String getParentDirectoryPath(CustomFile customFile) {
+            logger.warn("File does not exist: {}", filePath);
+            return false;
+        }
+
+        if (fileToDelete.isDirectory()) {
+            if (!deleteDirectory(fileToDelete)) {
+                logger.error("Failed to delete directory: {}", fileToDelete.getAbsolutePath());
+                return false;
+            }
+        } else {
+            if (!fileToDelete.delete()) {
+                logger.error("Failed to delete file: {}", fileToDelete.getAbsolutePath());
+                return false;
             }
+        }
+
+        // Delete file from repository
+        fileRepository.delete(file);
+        logger.info("File with ID {} deleted", id);
+        return true;
     }
 
-    public void deleteFile(Long id) {
-        Optional<CustomFile> fileOptional = fileRepository.findById(id);
-        if (fileOptional.isPresent()) {
-            fileRepository.delete(fileOptional.get());
-        } 
+    /**
+     * Deletes the specified directory and all its contents.
+     *
+     * @param directory The directory to delete.
+     * @return true if the directory was successfully deleted, false otherwise.
+     */
+    private boolean deleteDirectory(File directory) {
+        File[] files = directory.listFiles();
+        logger.info("Deleting directory: {}", directory.getAbsolutePath());
+        if (files != null) {
+            for (File file : files) {
+                if (file.isDirectory()) {
+                    if (!deleteDirectory(file)) {
+                        return false;
+                    }
+                } else {
+                    if (!file.delete()) {
+                        return false;
+                    }
+                }
+            }
+        }
+        return directory.delete();
+    }
+
+    /**
+     * Downloads the file with the specified name.
+     *
+     * @param fileId The id of the file to download.
+     * @return ResponseEntity with the file resource and headers.
+     */
+    public ResponseEntity<Resource> downloadFileById(Long fileId) throws MalformedURLException {
+        Optional<CustomFile> customFile = fileRepository.findById(fileId);
+        if (customFile.isEmpty()) {
+            return ResponseEntity.notFound().build();
+        }
+
+        String sFilePath = customFile.get().getPath();
+        String rootPath = System.getProperty("user.dir");
+        sFilePath = rootPath + sFilePath;
+
+        Path filePath = Paths.get(sFilePath);
+        Resource fileResource = new UrlResource(filePath.toUri());
+
+        if (fileResource.exists() && fileResource.isReadable()) {
+            HttpHeaders headers = new HttpHeaders();
+            headers.add(HttpHeaders.CONTENT_DISPOSITION, "attachment; filename=\"" + fileResource.getFilename() + "\"");
+
+            return ResponseEntity.ok()
+                    .headers(headers)
+                    .body(fileResource);
+        } else {
+            return ResponseEntity.notFound().build();
+        }
+    }
```

Comment:
I don't see a use for this endpoint... (unless it is being used for uploads? But it previously wasn't)